openwisp_network_topology/api/views.py

```diff
@@ -16,7 +24,34 @@
 Topology = swapper.load_model('topology', 'Topology')
 
 
-class NetworkCollectionView(generics.ListAPIView):
+class CustomDjangoModelPermissions(DjangoModelPermissions):
+    """
+    Add GET method permission check to DjangoModelPermissions.
+    """
+
+    def __init__(self):
+        self.perms_map = deepcopy(self.perms_map)
+        self.perms_map['GET'] = [
+            f'{Topology._meta.app_label.lower()}.view_{Topology._meta.object_name.lower()}'
+        ]
+
+
+class RequireAuthentication(APIView):
+    def __init__(self):
+        if app_settings.TOPOLOGY_API_AUTH_REQUIRED:
+            self.authentication_classes = [
+                SessionAuthentication,
+                BasicAuthentication,
+                BearerAuthentication,
+            ]
+            self.permission_classes = [
+                IsAuthenticated,
+                IsOrganizationMember,
+                CustomDjangoModelPermissions,
+            ]
```

Comment:
also, let's set these as class attributes (rather than setting them as instance attribute in the init method).

```python
class RequireAuthentication(APIView):
    if app_settings.TOPOLOGY_API_AUTH_REQUIRED:
        authentication_classes = [
            # ...
        ]
        permission_classes = [
            # ...
        ]
```